harness/tests/storage/test_s3.py

```diff
@@ -41,10 +47,48 @@ def live_manager(tmp_path: Path, require_secrets: bool) -> storage.S3StorageMana
 
 
 @pytest.mark.cloud  # type: ignore
-def test_live_s3_lifecycle(live_manager: storage.S3StorageManager, require_secrets: bool) -> None:
+@pytest.mark.parametrize(  # type: ignore
+    "prefix,storage_id,expected_storage_prefix,should_fail",
+    [
+        (None, "bar", "bar", False),
+        ("/", "bar", "bar", False),
+        ("///", "bar", "bar", False),
+        ("foo", "bar", "foo/bar", False),
+        ("/foo", "bar", "foo/bar", False),
+        ("foo/", "bar", "foo/bar", False),
+        ("/foo/", "bar", "foo/bar", False),
+        ("./foo/", "bar", "foo/bar", False),
+        ("/./foo/", "bar", "foo/bar", False),
+        ("/foo/..", "-", "-", True),
+    ],
+)
+def test_storage_prefix_normalization(
```

Comment:
Yup, good call. I can't even call the constructor of the S3StorageManager because it uses the network. I moved this to a helper function so that we can validate it without the network cost.